### PR TITLE
Added code for palindrome partitioning problem under dynamic programming

### PR DESCRIPTION
**Pull request by agnivg**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/palindrome_partitioning.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    >>> find_minimum_partitions("aab")
+    1
+    >>> find_minimum_partitions("aaa")
+    0
+    >>> find_minimum_partitions("ababbbabbababa")
+    3
+    """
+    n = len(string)
+    cut = [0 for i in range(n)]
```
Comment (CaedenPH) [inline dynamic_programming/palindrome_partitioning.py:24]:
```suggestion
    cut = [0] * n
```

**Review by CaedenPH — APPROVED** [review]
Small change

**Review by CaedenPH — APPROVED** [review]

File: dynamic_programming/palindrome_partitioning.py (hunk 1 above)
Comment (agnivg) [reply dynamic_programming/palindrome_partitioning.py]: I have fixed it in my new commit.

File: dynamic_programming/palindrome_partitioning.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    >>> find_minimum_partitions("aab")
+    1
+    >>> find_minimum_partitions("aaa")
+    0
+    >>> find_minimum_partitions("ababbbabbababa")
+    3
+    """
+    n = len(string)
+    cut = [0] * n
+    ispalindrome = [[False for i in range(n)] for j in range(n)]
```
Comment (CaedenPH) [inline dynamic_programming/palindrome_partitioning.py:25]:
```suggestion
    is_palindromic = [[False for i in range(n)] for j in range(n)]
```

File: dynamic_programming/palindrome_partitioning.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    for i in range(n):
+        mincut = i
+        for j in range(i + 1):
+            if string[i] == string[j] and (i - j < 2 or ispalindrome[j + 1][i - 1]):
+                ispalindrome[j][i] = True
```
Comment (CaedenPH) [inline dynamic_programming/palindrome_partitioning.py:30]:
```suggestion
                is_palindromic[j][i] = True
```

**Review by CaedenPH — COMMENTED** [review]
@cclauss Looks good

**Review by cclauss — APPROVED** [review]

**Pull request merged**